bgpd/bgp_io.c

```diff
@@ -173,12 +173,11 @@ static int read_ibuf_work(struct peer *peer)
 	uint16_t pktsize = 0;
 	struct stream *pkt;
 
-	/* Hold the I/O lock, we might not have space on the InQ */
-	frr_mutex_lock_autounlock(&peer->io_mtx);
```

Comment:
hmm - is it ok to be touching the ringbuf, for example, without the lock?

Reply:
yes.  Since (a) before the code change we did and (b) the only code that touches the ringbuf is on the bgp_io pthread

Reply:
ok, got it!